build/Publish.targets

```diff
@@ -65,4 +67,30 @@
       ContainerName="$(ArtifactContainerName)"
       Items="@(CliVersionBadgeToUpload)" />
   </Target>
+
+  <Target Name="PublishCliCoherentBadge"
+          Condition=" '$(Coherent)' == 'true' ">
+    <ItemGroup>
+      <CliCoherentBadgeToUpload Include="$(CoherentBadge)" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <CliCoherentBadgeToUpload>
+        <RelativeBlobPath>$(Product)/$(FullNugetVersion)/$([System.String]::Copy('%(Filename)%(Extension)').Replace('\' ,'/'))</RelativeBlobPath>
+      </CliCoherentBadgeToUpload>
+    </ItemGroup>
+
+    <UploadToAzure
```

Comment:
Can you piggyback on the code that uploads the VersionBadge itself instead of duplicating this here? Upload to Azure takes a list of items, you can just add the coherent badge there, when it is appropriate to publish a coherent badge.